salt/modules/win_snmp.py

```diff
@@ -33,17 +42,33 @@ def __virtual__():
     '''
     Only works on Windows systems.
     '''
-    if salt.utils.is_windows():
-        return __virtualname__
-    return False
+    if not salt.utils.is_windows():
+        return False, 'Module win_snmp: Requires Windows'
+
+    if not __salt__['reg.read_value'](_HKEY, _SNMP_KEY)['success']:
+        return False, 'Module win_snmp: SNMP not installed'
+
+    return __virtualname__
+
+
+def _to_unicode(instr):
+    '''
+    Converts from current users character encoding to unicode.
+    When instr has a value of None, the return value of the function
+    will also be None.
+    '''
+    if instr is None or isinstance(instr, six.text_type):
+        return instr
+    else:
+        return six.text_type(instr, 'utf8')
```

Comment:
@s0undt3ch It needs to be utf8 to work properly. `__salt_system_encoding__` uses just what it says, the system encoding. `salt.utils.to_unicode` does not output the needed Unicode characters.

For example, if I send this as one of the parameters: `tm™-forkψ-euro€`, the function receives it as a string:
```
tm™-forkψ-euro€
<type 'str'>
```
When I pass that to `salt.utils.to_unicode`, it is returned as:
```
tmâ„¢-forkÏˆ-euroâ‚¬
<type 'unicode'>
```
When I pass it to the local `_to_unicode` function, I get what I need:
```
tm™-forkψ-euro€
<type 'unicode'>
```
If the `salt.utils.to_unicode` function is supposed to output like the last one, then it's not working properly... and we should fix it and use it instead of the local `_to_unicode`. I just didn't want to change it and break a hundred things somewhere else.